src/Uno.APISurfaceValidator/Program.cs

```diff
@@ -0,0 +1,183 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Transactions;
+using Mono.Cecil;
+using Mono.Collections.Generic;
+
+namespace Uno.APISurfaceValidator
+{
+	class Program
+	{
+		private static (string path, string targetAssembly, string pattern)[] _contractsToValidate = new[]
+		{
+			(@"C:\Program Files (x86)\Windows Kits\10\References\10.0.19041.0\Windows.Foundation.FoundationContract\4.0.0.0\Windows.Foundation.FoundationContract.winmd", "Uno.Foundation.dll", "*"),
+			(@"C:\Program Files (x86)\Windows Kits\10\References\10.0.19041.0\Windows.Foundation.UniversalApiContract\10.0.0.0\Windows.Foundation.UniversalApiContract.winmd", "Uno.dll", "-Windows.UI.Xaml"),
+			(@"C:\Program Files (x86)\Windows Kits\10\References\10.0.19041.0\Windows.Phone.PhoneContract\1.0.0.0\Windows.Phone.PhoneContract.winmd", "Uno.dll", "*"),
+			(@"C:\Program Files (x86)\Windows Kits\10\References\10.0.19041.0\Windows.Networking.Connectivity.WwanContract\2.0.0.0\Windows.Networking.Connectivity.WwanContract.winmd", "Uno.dll", "*"),
+			(@"C:\Program Files (x86)\Windows Kits\10\References\10.0.19041.0\Windows.ApplicationModel.Calls.CallsPhoneContract\5.0.0.0\Windows.ApplicationModel.Calls.CallsPhoneContract.winmd", "Uno.dll", "*"),
+			(@"C:\Program Files (x86)\Windows Kits\10\References\10.0.19041.0\Windows.Foundation.UniversalApiContract\10.0.0.0\Windows.Foundation.UniversalApiContract.winmd", "Uno.UI.dll", "+Windows.UI.Xaml")
+		};
+
+		static int Main(string[] args)
+		{
+			var hasErrors = false;
+			var filePath = UnpackArchive(args[0]);
+
+			Console.WriteLine($"Validating package {args[0]}");
+
+			foreach (var contract in _contractsToValidate)
+			{
+				Console.WriteLine($"Validating {Path.GetFileName(contract.path)} :");
+
+				var referenceAssembly = ReadAssemblyDefinition(contract.path);
+				var assembly = ReadAssemblyDefinition(Path.Combine(filePath, "lib", "netstandard2.0", contract.targetAssembly));
```

Comment:
What about Android, iOS, macOS?

Reply:
Not yet, will be added in subsequent phases